integration-cli/docker_cli_by_digest_test.go

```diff
@@ -403,7 +403,12 @@ func (s *DockerRegistrySuite) TestDeleteImageByIDOnlyPulledByDigest(c *check.C)
 		c.Fatalf("error inspecting image id: %v", err)
 	}
 
-	dockerCmd(c, "rmi", imageID)
+	if out, _, _ := dockerCmdWithError("rmi", imageID); !strings.Contains(out, "(must be forced) - image is referenced in one or more repositories") {
+		c.Fatalf("expected error removing image by ID while there was a repo digest reference. Output:\n%s", out)
+	}
+
+	// Forcing it should work.
+	dockerCmd(c, "rmi", "-f", imageID)
```

Comment:
The existing repository digest reference prevents the image from being deleted by ID unless forced.